### ISSUE TITLE
`padded-blocks`: support for interface & type

### ISSUE THREAD

**Issue by timnieder** [eslint-stylistic/eslint-stylistic#427]
### Clear and concise description of the problem

The padded-blocks rule has currently no support for typescript constructs like interface and type.
This code will not be checked with the current rule:
```ts
type Type = {

  s: string;

};

interface Interface {

  s: string;

}
```

I haven't found any alternative rule, but one could probably do some tricks with `padding-line-between-statements`

### Suggested solution

The rule `padded-blocks` should support interfaces and types, preferably customizable through the object option.

### Validations

- [X] Follow our [Code of Conduct](https://github.com/eslint-stylistic/eslint-stylistic/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guide](https://eslint.style/contribute/guide).
- [X] Check that there isn't already an issue that request the same feature to avoid creating a duplicate.

### Contributes

- [ ] If this feature request is accepted, I am willing to submit a PR to fix this issue

**Comment by 9romise** [comment]
If this is accepted, I'm willing to submit a PR after confirming some details and making changes in #308 .

**Comment by That-Guy977** [comment]
This should probably be handled by `object-curly-newline` instead.

**Comment by Hydradox** [comment]
Im interested in this feature, has any progress being made?
